### PR TITLE
chore: releases with Goreleaser and Github Actions

### PR DESCRIPTION
**Pull request by sigua-cs**
Add releases and docker images using Goreleaser running on Github Actions

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          DOCKER_USERNAME: ${{ secrets.docker_username }}
+          DOCKER_PASSWORD: ${{ secrets.docker_password }}
+
+      -
+        name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          distribution: goreleaser
+          version: latest
+          args: release --rm-dist --debug
```
Comment (vfoucault) [inline .github/workflows/release.yml]: not sure that `--rm-dist` is needed here

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+builds:
+  - binary: chproxy
+    goos:
+      - linux
+    goarch:
+      - amd64
```
Comment (vfoucault) [inline .goreleaser.yml:10]: Why not building also for more Archs and OS ?

Comment (sigua-cs) [reply .goreleaser.yml:10]: We plan to do it. But first, we should test cross-compilation using Github Actions

File: go.mod [hunk 3]
```diff
@@ -3,7 +3,7 @@ module github.com/contentsquare/chproxy
 go 1.13
```
Comment (gontarzpawel) [inline go.mod]: I think you should bump the version to the supplied in docker image:
```suggestion
go 1.17
```
You'll have to run after that 
```
go mod tidy
```

Tested in local, works fine

File: .goreleaser.yml (hunk 2 above)
Comment (sigua-cs) [reply .goreleaser.yml:10]: I tested another ARCHs, and the best way to compile additional ARCHs is to have a dedicated build for each arch separately. So, I will add one by one in the next commits

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+      -
+        name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Install cross-compile package
+        run: |
+          sudo apt update
```
Comment (sigua-cs) [inline .github/workflows/release.yml:29]: It's good practice to update before dependencies installation, but I'm not sure if we really need it

File: .github/workflows/release.yml (hunk 1 above)
Comment (sigua-cs) [reply .github/workflows/release.yml]: Indeed. It was used for Goreleaser build debugging.

**Review by gontarzpawel — APPROVED** [review]

**Pull request merged**